### DISCUSSION TITLE
Unable to find function

### DISCUSSION THREAD

**Discussion by Emmanuel-324** [idaholab/moose#28991] [Q&A Modules: Solid mechanics]
Hey, a quick question. I have this auxkernel block that I want to implement a function that works as the comment stated. 
```
[temperature]
      type = FunctionAux
      variable = temperature
      function = 't < 1 ? 600 : 600' # temperature jumps to 600 at the start and stays constant
      execute_on = timestep_begin
    []
```
But I get this error
```
Unable to find function t < 1 ? 600: 600
```

How do I resolve this issue?

**Comment by GiudGiud** [comment]
Hello

You can't specify a parsed function directly in a function parameter

You have to create a ParsedFunction in the Functions block

**Reply by Emmanuel-324** [reply]
I modifed it this way 
```
[Functions]
  [temperature]
    type = ParsedFunction
    value = 't < 1 ? 300 : 300' # temperature jumps to 600 at the start and stays constant
    execute_on = 'timestep_begin'
  []
[] 
```
And I got an error in the parsed function
```
ERROR: FunctionParser is unable to parse expression: t < 1 ? 300 : 300
Syntax error: Operator expected
[0] /home/emmanuel324/projects/moose/scripts/../libmesh/installed/include/libmesh/parsed_function.h, line 564, compiled Aug 12 2024 at 18:26:31
:
ERROR: FunctionParser is unable to parse expression: t < 1 ? 300 : 300
Syntax error: Operator expected
[1] /home/emmanuel324/projects/moose/scripts/../libmesh/installed/include/libmesh/parsed_function.h, line 564, compiled Aug 12 2024 at 18:26:31
```

**Reply by GiudGiud** [reply]
You need to set the parameter use_xyzt iirc

Also the if syntax is wrong. You're writing it like c++ but if should be 

if(a, b , c)
With a the condition, b the true, and c the false

**Reply by Emmanuel-324** [reply]
I can modify the if statement as:

value = 'if(t < 1, 300, 300)' # temperature jumps to 300 at t = 1 and stays constant

But I do not understand what you mean by
```
You need to set the parameter use_xyzt iirc
```

**Reply by GiudGiud** [reply]
That's a parameter that is used for parsed expressions. ParsedFunction might not be using it

**Reply by Emmanuel-324** [reply]
And how do I set that parameter if you do not mind

**Reply by GiudGiud** [reply]
like this
```
use_xyzt=true
```
but I just checked and ParsedFunction does not use this parameter. It's for ParsedMaterial
